New touchpad-like cursor control with acceleration support

prulitarit opened the pull request.

prulitarit committed:
Add support for capacitive HOME button

prulitarit committed:
New touchpad-like cursor control with acceleration support

Please, edit MtTranslate.hpp "Acceleration factor and distance" for your best experience.

prulitarit committed:
fixes

prulitarit committed:
Tap and hold for right button click function

prulitarit committed:
Fixes

prulitarit committed:
fixes

prulitarit committed:
fixes

jjackowski commented:
I'm working on parsing command line arguments to allow the program to act differently depending on how it is started. Specifically, I do want absolute mouse positioning, but I don't want to remove your relative motion. While doing that, I noticed that your implementation still makes an input device that presents itself as providing the absolute mouse location instead of relative motion. I suppose that would make sense if you needed to control that acceleration value so that it can be different than what is used with a mouse or the system's configuration. Was that your intent? Otherwise it may make more sense if EvdevOutput provides REL_[XY] instead of ABS_[XY].

jjackowski commented:
In MtTranslate.cpp, you can use this code instead of making Evdev::fd public. It really shouldn't be public.

// get digitizer resolution
const input_absinfo *ai;
ai = evdev->absInfo(ABS_X);
if (ai->maximum) {
 	 maxX = ai->maximum; // - ai->minimum; // minimum might not be zero
 	 cursorX = maxX / 2;
}

prulitarit commented:
> I'm working on parsing command line arguments to allow the program to act differently depending on how it is started. Specifically, I do want absolute mouse positioning, but I don't want to remove your relative motion. While doing that, I noticed that your implementation still makes an input device that presents itself as providing the absolute mouse location instead of relative motion. I suppose that would make sense if you needed to control that acceleration value so that it can be different than what is used with a mouse or the system's configuration. Was that your intent? Otherwise it may make more sense if EvdevOutput provides REL_[XY] instead of ABS_[XY].

Yes you are right. I just haven't found any other way to implement this behavior. thanks for the hint.

of course it would be nice if the device could register as one that provides relative motion and respect the system acceleration settings.

jjackowski commented:
> of course it would be nice if the device could register as one that provides relative motion and respect the system acceleration settings.

A regular common mouse provides REL_X and REL_Y. Any mouse cursor acceleration is handled elsewhere, not as low-level as the input device.

prulitarit committed:
Show/hide onboard by Home button and switch tapping mode

Show/hide on-screen keyboard 'onboard' when tapping on "Home" (Windows) button, found on most tablets and switch mode between instant click/touchpad-like click. Onboard should be installed. This will make your Ubuntu tablet as easy to use as Windows 10

prulitarit committed:
fixes

prulitarit committed:
fixes
